src/RamRelation.h

```diff
@@ -314,6 +322,86 @@ class RamRelation {
         num_tuples++;
     }
 
+    /** insert a new tuple to table, possibly more than one tuple depending on relation type */
+    void insert(const RamDomain* tuple) {
+        // TODO: (pnappa) an eqrel check here is all that appears to be needed for implicit additions
+        if (id.isEqRel()) {
+            // TODO: future optimisation would require this as a member datatype
+            // brave soul required to pass this quest
+            // // specialisation for eqrel defs
+            // std::unique_ptr<binaryrelation> eqreltuples;
+            // in addition, it requires insert functions to insert into that, and functions
+            // which allow reading of stored values must be changed to accommodate.
+            // e.g. insert =>  eqRelTuples->insert(tuple[0], tuple[1]);
+
+            // for now, we just have a naive & extremely slow version, otherwise known as a O(n^2) insertion
+            // ):
+
+            // store all values that will be implicitly relevant to the two that we will insert
+            std::list<const RamDomain*> relevantStored;
```

Comment:
After benchmarking - it does appear that vector comes out as winner:

```
vec
Time taken to insert: 751.987ms
dtor time: 40.147ms
list
Time taken to insert: 7852.48ms
dtor time: 11771.9ms
blocklist
Time taken to insert: 6270.67ms
dtor time: 88.1499ms
```

Switched it over now. However, BlockList does have a place still in the rest of the code for storing atomics, it appears (copy/move semantics and block allocation superiority).